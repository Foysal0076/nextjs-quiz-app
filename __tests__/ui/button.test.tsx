import '@testing-library/jest-dom'

import { render, screen } from '@testing-library/react'
import React from 'react'

import Button from '@/components/ui/button'

describe('Button', () => {
  it('renders with default props', () => {
    render(<Button>Click me</Button>)
    const button = screen.getByRole('button', { name: /click me/i })
    expect(button).toBeInTheDocument()
    expect(button).toHaveClass('bg-primary-500')
  })

  it('renders with custom variant and size', () => {
    render(
      <Button variant='secondary' size='lg'>
        Custom Button
      </Button>
    )
    const button = screen.getByRole('button', { name: /custom button/i })
    expect(button).toHaveClass('bg-none')
    expect(button).toHaveClass('h-[3rem]')
  })

  it('renders in loading state', () => {
    render(<Button loading>Loading</Button>)
    expect(screen.getByRole('button')).toBeDisabled()
    expect(screen.getByTestId('spinner')).toBeInTheDocument()
  })

  it('applies custom className', () => {
    render(<Button className='custom-class'>Custom Class Button</Button>)
    const button = screen.getByRole('button', { name: /custom class button/i })
    expect(button).toHaveClass('custom-class')
  })

  it('matches snapshot', () => {
    const { container } = render(<Button>Click me</Button>)
    expect(container).toMatchSnapshot()
  })
})
