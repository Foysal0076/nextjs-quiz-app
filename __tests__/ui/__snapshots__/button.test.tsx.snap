// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    class="relative select-none rounded-lg transition-all duration-200 text-nowrap focus:outline-none font-medium bg-primary-500 hover:bg-primary-600 focus:ring-4 focus:ring-primary-100 focus:outline-primary-200 disabled:active:transform-none active:scale-[.95] disabled:text-neutral-200 disabled:bg-primary-100 disabled:cursor-not-allowed text-neutral-0 dark:text-neutral-900 h-[2.5rem] text-md px-8"
  >
    <div
      class="visible"
    >
      Click me
    </div>
  </button>
</div>
`;
